apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/common/db/SearchQueryParser.java

```diff
@@ -0,0 +1,21 @@
+package edu.pw.apsienrollment.common.db;
+
+import lombok.NonNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class SearchQueryParser {
+    public static Collection<SearchCriteria> parse(@NonNull String searchQuery) {
+        Pattern pattern = Pattern.compile("([\\w.]+?)(:|>|>=|<|<=)((\\w|:|-|;|\\p{Space})+?),", Pattern.UNICODE_CHARACTER_CLASS);
```

Comment:
`:` is allowed both as an operator and as a value fragment, is it desired behaviour?

Reply:
this can be changed. Operator could be "=" however this change will affect client app. Is that ok?

Reply:
As far as i know there is no API module in client yet, so it won't affect client at all.